File: backend/src/main/java/isep/webapp/terasync/model/odSystem/CustomerOrderList.java
package isep.webapp.terasync.model.odSystem;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class CustomerOrderList extends isep.webapp.terasync.model.Entity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customerOrderListId")
    private int customerOrderListId;
    @CreationTimestamp
    @Column(name = "selectionTime")
    private String selectionTime;
    @Column(name = "quantity")
    private int quantity;
    @Column(name = "customerOrderId")
    private int customerOrderId;
    @Column(name = "productId")
    private int productId;
}

File: front/src/app/model/odSystem/customer.order.list.ts
import {Product} from "./product";

export class CustomerOrderList {
  customerOrderListId: number | undefined
  selectionTime: string | undefined;
  quantity: number;
  customerOrderId: number;
  productId: number;

  product: Product | undefined;

  constructor(productId: number, quantity: number, customerOrderId: number, selectionTime?: string, customerOrderListId?: number) {
    this.customerOrderListId = customerOrderListId;
    this.selectionTime = selectionTime;
    this.quantity = quantity;
    this.customerOrderId = customerOrderId;
    this.productId = productId;
  }

  static fromJson(jsonCustomerOrderList: CustomerOrderList) {
    let customerOrderList = new CustomerOrderList(jsonCustomerOrderList.productId, jsonCustomerOrderList.quantity, jsonCustomerOrderList.customerOrderId, jsonCustomerOrderList.selectionTime, jsonCustomerOrderList.customerOrderListId);

    customerOrderList.product = Product.fromJson(jsonCustomerOrderList.product!);

    return customerOrderList;
  }

  static initializeCustomerOrderLists(json: {customerOrderLists: CustomerOrderList[]}) {
    let customerOrderLists: CustomerOrderList[] = [];
    if (json.customerOrderLists) {
      for (let customerOrderList of json.customerOrderLists) {
        customerOrderLists.push(CustomerOrderList.fromJson(customerOrderList));
      }
    }
    return customerOrderLists;
  }
}
